### PR TITLE
Fix/13883 mask admission states design requirements

### PR DESCRIPTION
**Pull request by flxschmidt**
## Description
- Fix the layout behavior for mask state and admission state in certificate overview
- Fix the layout behavior for admission state in person detail

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13883

## Screenshots
Screen Shot 2022-09-14 at 16 47 16

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift [hunk 1]
```diff
@@ -239,3 +242,49 @@ class HealthCertifiedPersonCellModel {
...
+		/// Show Mask Status Badge with 80% width
+		case maskStatusVisibleAdmissionStatusNull
+		
+		/// Show Spacer with 100% width
+		case maskStatusNullAdmissionStatusNull
+	}
+	
+	/// Returns how to configure the view for admission states:
+	/// Mask Status and Admission Status
+	var admissionStatesConfiguration: MaskAndAdmissionStatesConfiguration {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift:272]:
```suggestion
	var maskAndAdmissionStatesConfiguration: MaskAndAdmissionStatesConfiguration {
```

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift [hunk 2]
```diff
@@ -239,3 +242,49 @@ class HealthCertifiedPersonCellModel {
...
+		case maskStatusInvisibleAdmissionStatusNull
+		
+		/// Show Mask Status Badge with 80% width
+		case maskStatusVisibleAdmissionStatusNull
+		
+		/// Show Spacer with 100% width
+		case maskStatusNullAdmissionStatusNull
+	}
+	
+	/// Returns how to configure the view for admission states:
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift:270]:
```suggestion
	/// Returns how to configure the view for mask and admission states
```

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift [hunk 3]
```diff
@@ -429,6 +427,57 @@ class HealthCertifiedPersonTableViewCell: UITableViewCell, ReuseIdentifierProvid
 
 		addGestureRecognizer(tapGestureRecognizer)
 	}
+	
+	private func configureAdmissionStatesStackView() {
+		guard let admissionStatesConfiguration = cellModel?.admissionStatesConfiguration else {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift:432]:
```suggestion
		guard let maskAndAdmissionStatesConfiguration = cellModel?.admissionStatesConfiguration else {
```

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift [hunk 4]
```diff
@@ -89,8 +90,10 @@ class HealthCertifiedPersonTableViewCell: UITableViewCell, ReuseIdentifierProvid
 			accessibilityIdentifier: AccessibilityIdentifiers.HealthCertificate.MaskState.roundedView,
 			labelAccessibilityIdentifier: AccessibilityIdentifiers.HealthCertificate.MaskState.title
 		)
-		 
-		maskAdmissionStatesView.isHidden = !cellModel.isShortAdmissionStatusVisible && !cellModel.isMaskStatusVisible
+		
+		maskAdmissionStatesStackView.isHidden = !cellModel.isShortAdmissionStatusVisible && !cellModel.isMaskStatusVisible
+		configureAdmissionStatesStackView()
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift:95]:
```suggestion
		configureStatesStackView()
```

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift [hunk 5]
```diff
@@ -429,6 +427,57 @@ class HealthCertifiedPersonTableViewCell: UITableViewCell, ReuseIdentifierProvid
...
+	
+	private func configureAdmissionStatesStackView() {
+		guard let admissionStatesConfiguration = cellModel?.admissionStatesConfiguration else {
+			maskAdmissionStatesStackView.isHidden = true
+			return
+		}
+		
+		[maskStateView, admissionStateView, admissionStatesSpacerView].forEach { $0.isHidden = true }
+
+		switch admissionStatesConfiguration {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift:439]:
```suggestion
		switch maskAndAdmissionStatesConfiguration {
```

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift [hunk 6]
```diff
@@ -239,3 +242,49 @@ class HealthCertifiedPersonCellModel {
 		}
 	}
 }
+
+extension HealthCertifiedPersonCellModel {
+	enum MaskAndAdmissionStatesConfiguration {
+		/// Show Nothing
+		case maskStatusInvisibleAdmissionStatusInvisible
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift:249]: Can we rename every where maskStatus to maskState and admissionStatus to admissionState ?

**Review by puneetmahali1 — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift [hunk 7]
```diff
@@ -188,17 +191,26 @@ class HealthCertifiedPersonTableViewCell: UITableViewCell, ReuseIdentifierProvid
...
 		return stackView
 	}()
 
 	private lazy var admissionStateView = RoundedLabeledView()
+	
 	private lazy var maskStateView = RoundedLabeledView()
-
-	private lazy var maskAdmissionStatesView = UIView()
+	
+	private lazy var admissionStatesSpacerView = UIView()
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift:205]:
```suggestion
	private lazy var emptyStateSpacerView = UIView()
```

**Review by naveeddotio — CHANGES_REQUESTED** [review]
Overall, I really liked the approach, however I have some recommendations regarding variable names.

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift (hunk 1 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift]: 👍

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift (hunk 2 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift]: 👍

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift (hunk 3 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift]: 👍

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift (hunk 4 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift]: 👍

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift (hunk 5 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift]: 👍

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift (hunk 6 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonCellModel.swift]: 👍

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift (hunk 7 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift]: 👍

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**